doc: flesh out user credential documentation (#15956)
sourcegraph/sourcegraph: doc/campaigns/how-tos/configuring_user_credentials.md

# Configuring user credentials

> NOTE: This page describes functionality added in Sourcegraph 3.22. Older Sourcegraph versions only allow campaigns to be applied and managed by site admins.

> Before using campaigns, you need to add a personal access token for each code host that your campaigns interact with. These tokens are used by Sourcegraph to create and manage changesets as you, and with your specific permissions.

Reviewer: ```suggestion
In order to [publish changesets with campaigns](publishing_changesets.md), you need to add a personal access token for each code host that your campaigns interact with. These tokens are used by Sourcegraph to create and manage changesets as you, and with your specific permissions, on the code host.
```

> > NOTE: The only type of user credential supported by Sourcegraph right now is a personal access token. Further credential types may be supported in the future.

Author: Yeah, that's a good call. Moving to known issues.

## Requirements

- Sourcegraph instance with repositories in it. See the "[Quickstart](../../index.md#quickstart)" guide on how to setup a Sourcegraph instance.

## Adding a personal access token

> Adding personal access tokens is done through the the Campaigns section of your user settings. To access this page, follow these instructions (also shown in the video below):

Reviewer: ```suggestion
Adding personal access tokens is done through the the Campaigns section of your user settings:
```
